### DISCUSSION TITLE
[Snaps] Custom UI improvements - image and input fields

### DISCUSSION THREAD

**Discussion by golobitch** [MetaMask/snaps#1285] [Ideas]
I am working on a snap for XRPL and it is coming quite nice together with all the basic functionalities.

I know that for know, there is limited set of components that we can use in Snaps.

However, I am missing some very useful elements in snaps.

For example, I would like to have option to show image (NFT), when placing an order to buy it.

![image](https://user-images.githubusercontent.com/24848451/226115545-c1881827-9f7e-4628-817e-df7a6371e873.png)

Also, it would be nice to specify more than just one input field (with prompt dialog).  For now, I am circumventing this through multiple dialogs as a steps (first dialog, step 1, second dialog step 2, etc...)

I would like to do it in one dialog like this.
![image](https://user-images.githubusercontent.com/24848451/226115658-b3267d4f-811d-49dc-be11-3869a53388ab.png)

Is this something that you would consider doing or maybe do you already have this on your roadmap?

**Comment by therealharpaljadeja** [comment]
Are these mockups or implemented?

**Reply by golobitch** [reply]
Those two empty input fields and that box that says NFT Image are mockups, everything else is implement

**Comment by Montoya** [comment]
We will eventually support this. It is on our roadmap. But not planned as part of our first release to the MetaMask extension in August.

**Reply by golobitch** [reply]
So we can expect that MetaMask extension will get support for Snaps in August? :)

**Reply by Montoya** [reply]
Yes, our current target date to launch Snaps in the extension is August 16

**Reply by piotr-dziubecki** [reply]
> Yes, our current target date to launch Snaps in the extension is August 16

Fingers crossed! We're already testing it on the Casper blockchain.

**Reply by therealharpaljadeja** [reply]
So is it you guys doing it along with the Metamask team or you modified Flask to support images? 😅
